### PR TITLE
[mypy] Fix `matrix_operation.py`

### PR DESCRIPTION
**Pull request by Rohanrbharadwaj**
### Describe your change:

Related to #4052 
Replacement for #5806 

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by Rohanrbharadwaj** [comment]
@cclauss Your reviews, please.

File: matrix/matrix_operation.py [hunk 1]
```diff
@@ -18,6 +18,7 @@ def add(*matrix_s: list[list]) -> list[list]:
         for i in matrix_s[1:]:
             _verify_matrix_sizes(matrix_s[0], i)
         return [[sum(t) for t in zip(*m)] for m in zip(*matrix_s)]
+    raise TypeError("Expected a matrix, got int/list instead")
```
Comment (cclauss) [inline matrix/matrix_operation.py:21]: Could we please have a doctest that proves this error will be raised?  It should look something like this:
```python
>>> add(<something_problematic_here>)
Traceback (most recent call last):
  ...
TypeError: Expected a matrix, got int/list instead
```

**Review by cclauss — CHANGES_REQUESTED** [review]
Doctests for the new exceptions please.

File: matrix/matrix_operation.py (hunk 1 above)
Comment (Rohanrbharadwaj) [reply matrix/matrix_operation.py:21]: Alright, on it

File: matrix/matrix_operation.py [hunk 2]
```diff
@@ -138,21 +149,23 @@ def inverse(matrix: list[list]) -> list[list]:
...
         for row in range(len(matrix))
     ]
-    adjugate = transpose(cofactors)
+    adjugate = list(transpose(cofactors))
     return scalar_multiply(adjugate, 1 / det)
 
 
 def _check_not_integer(matrix: list[list]) -> bool:
     if not isinstance(matrix, int) and not isinstance(matrix[0], int):
         return True
```
Comment (cclauss) [inline matrix/matrix_operation.py:158]:
```suggestion
    return not isinstance(matrix, int) and not isinstance(matrix[0], int)
```

File: matrix/matrix_operation.py [hunk 3]
```diff
@@ -138,21 +149,23 @@ def inverse(matrix: list[list]) -> list[list]:
...
-    raise TypeError("Expected a matrix, got int/list instead")
+    return False
```
Comment (cclauss) [inline matrix/matrix_operation.py:159]:
```suggestion
```

File: matrix/matrix_operation.py [hunk 4]
```diff
@@ -91,6 +101,7 @@ def transpose(matrix: list[list], return_map: bool = True) -> list[list]:
             return map(list, zip(*matrix))
         else:
             return list(map(list, zip(*matrix)))
+    raise TypeError("Expected a matrix, got int/list instead")
```
Comment (cclauss) [inline matrix/matrix_operation.py:104]: One more doctest for this exception, please.

**Review by cclauss — CHANGES_REQUESTED** [review]
Just a bit more...

**Review by cclauss — APPROVED** [review]
AWESOME!!!  Thanks much.

**Pull request merged**